lua/autorun/server/sv_daily_restart.lua

```diff
@@ -1,67 +1,88 @@
 
-    function alertClientsOfServerRestart()
-        print("Alerting clients of Restart")
-        restartAt = os.time() + (warningTime * 60)
+local function CFCDailyRestart:restartServer()
+    print("restarting server")
+    
+    file.Write(fileName..".txt", os.time())
 
-        -- TODO: Alert how many seconds left
-        -- You'll want a logarithmic-ish function to alert at:
-        -- 15 min, 10min, 5min, 4min, 3min,2min,1min,30s,15s,10,9,8.. so on
-        sendAlertToClients()
+    --local params = {}
+    --local headers = {}
+    --http.Post("localhost:2327/gmod/restart", params, function() end, function() end, headers)
+end
 
-        net.Start("AlertUsersOfRestart")
-            net.WriteFloat(restartAt)
-        net.Broadcast()
+local function CFCDailyRestart:sendAlertToClients(message)
+    for k, v in pairs(player.GetHumans()) do
+        v:ChatPrint("[CFC Daily Restart] "..message)
     end
+end
+
+local function CFCDailyRestart:alertClientsOfServerRestart()
+    print("Alerting clients of Restart")
+    CFCDailyRestart.restartAt = os.time() + (warningTime * 60)
+
+    CFCDailyRestart:sendAlertToClients("Restarting server in "..CFCDailyRestart.warningTime.." minutes.")
+
+    net.Start("AlertUsersOfRestart")
+        net.WriteFloat(restartAt)
+    net.Broadcast()
+end
 
-    --TODO: Create a timer to 
+timer.Create("CFC_DailyRestart", 1, 0, function()
+    local currentTime = os.time()
+    local lastScheduledRestart = CFCDailyRestart.lastRestart
+    local isLastScheduledRestartValid = not lastScheduledRestart == nil
+    local scheduledTimeThreshold = 3600 --3600 = 1 hour
 
-    timer.Create("CFC_DailyRestart", 1, 0, function()
-        -- TODO: Check to make sure that the latest restart time wasn't recently
-        -- if countingDownToRestart then return
-        -- if waitingForServerToEmpty then return
+    if isLastScheduledRestartValid then
+        if os.difftime(os.time(), lastScheduledRestart) <= scheduledTimeThreshold then return end
+    end
+
+    local formattedTime = os.date("*t", currentTime)
 
-        theTime = os.date("*t", os.time())
+    -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
+    --
+    -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
 
-        -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
-        --
-        -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
+    if not countingDownToRestart then
```

Comment:
Another code smell is a conditional that has many lines in it. Indicates that maybe this could be broken out into another function and then have a short-circuit here instead.